File: codewars/Anagram difference/anagram_difference.js
function anagramDifference(w1, w2) {
    if (w1 === "" && w2 === "") {
      return 0;
    } else if (w1 === "" || w2 === "") {
      if (w1 !== "") {
        return w1.length;
      } else {
        return w2.length;
      }
    }
    let letterCount = {};
    if (w1 !== "" && w2 !== "") {
      for (let letter of w1) {
        letterCount[letter] = (letterCount[letter] || 0) + 1;
      }
  
      for (let letter of w2) {
        if (letterCount[letter]) {
          letterCount[letter]--;
        } else {
          letterCount[letter] = -1;
        }
      }
    }
    let lettersToRemove = 0;
  
    for (let count of Object.values(letterCount)) {
      lettersToRemove += Math.abs(count);
    }
  
    return lettersToRemove;
  }

File: codewars/Anagram difference/Anagram_difference.c
#include <stdlib.h>

unsigned anagram_difference(const char *word1, const char *word2) {
    int A[26] = {}, B[26] = {};
    int i = 0;
    while (word1[i] != '\0') {
        A[word1[i] - 'a']++;
        i++;
    }
    i = 0;
    while (word2[i] != '\0') {
        B[word2[i] - 'a']++;
        i++;
    }
    unsigned a = 0;
    for (int i = 0; i < 26; i++) a += abs(A[i] - B[i]);
    return a;
}
